packages/daimo-api/src/router.ts

```diff
@@ -272,28 +273,33 @@ export function createRouter(
       .mutation(async (opts) => {
         const { name, pubKeyHex } = opts.input;
 
-        // Don't deploy the account; just get the counterfactual address
-        console.log(
-          `[API] not deploying account for ${name}, pubkey ${pubKeyHex}`
-        );
-        const account = await DaimoAccount.init(
-          l2Client,
-          pubKeyHex,
-          () => {
-            throw new Error("No signer");
-          },
-          false
-        );
-        const address = account.getAddress();
+        const address = await accountFactory.getAddress(pubKeyHex);
+
+        // TODO: Should be able to batch these in a single tx for perf.
+        // Promise.all awaiting doesn't work because nonces go out of sync.
```

Comment:
Waiting on these three txes makes onboarding quite slow, we should do this asap a follow up PR. Ironically it seems the "right way" to do this is to use an AA wallet for our backend as well lol.

Reply:
three other options:
1. **quick and dirty concurrency** - just kick off all 3 transactions in parallel, instead of one after the other. `await Promise.all` on the three transactions, then again for the three receipts.
1. **AccountFactory.deployAndClaimName** - a single function in AccountFactory that deploys and account, claims a name, and uses `value` (if passed, testnet only, unnecessary when we have a paymaster setup) to prefund with the EntryPoint
2. **deploy from account itself** - DaimoAccount initialized with a name, registers itself with the NameRegistry in first transaction. a few ways to get this transaction onchain, but the simplest would be to bundle it ourselves